testsupport/tiers/checks.go

```diff
@@ -1054,20 +1054,25 @@ func networkPolicyIngressFromPolicyGroup(name, group string) namespaceObjectsChe
 	return networkPolicyIngress(name, "network.openshift.io/policy-group", group)
 }
 
-func networkPolicyIngress(name, labelName, labelValue string) namespaceObjectsCheck {
+func networkPolicyIngress(name string, labelNameValuePairs ...string) namespaceObjectsCheck {
 	return func(t *testing.T, ns *corev1.Namespace, memberAwait *wait.MemberAwaitility, userName string) {
+		require.True(t, len(labelNameValuePairs)%2 == 0, "labelNameValuePairs must be a list of key-value pairs")
 		np, err := memberAwait.WaitForNetworkPolicy(t, ns, name)
 		require.NoError(t, err)
 		assert.Equal(t, toolchainv1alpha1.ProviderLabelValue, np.ObjectMeta.Labels[toolchainv1alpha1.ProviderLabelKey])
+
+		peers := []netv1.NetworkPolicyPeer{}
+		for labelNameValuePairsIndex := 0; labelNameValuePairsIndex < len(labelNameValuePairs); labelNameValuePairsIndex += 2 {
+			labelName := labelNameValuePairs[labelNameValuePairsIndex]
+			labelValue := labelNameValuePairs[labelNameValuePairsIndex+1]
+			peers = append(peers, netv1.NetworkPolicyPeer{NamespaceSelector: &metav1.LabelSelector{MatchLabels: map[string]string{labelName: labelValue}}})
+		}
+
 		expected := &netv1.NetworkPolicy{
 			Spec: netv1.NetworkPolicySpec{
 				Ingress: []netv1.NetworkPolicyIngressRule{
 					{
-						From: []netv1.NetworkPolicyPeer{
-							{
-								NamespaceSelector: &metav1.LabelSelector{MatchLabels: map[string]string{labelName: labelValue}},
-							},
-						},
+						From: peers,
```

Comment:
This doesn't match what is in the template. 
In the template you define two Ingress rules:
```yaml
  spec:
    ingress:
    - from:
      - namespaceSelector:
          matchLabels:
            kubernetes.io/metadata.name: openshift-virtualization-os-images
    - from:
      - namespaceSelector:
          matchLabels:
            kubernetes.io/metadata.name: openshift-cnv
```
while here you expect two label selectors for the same rule. that would correspond to:
```
  spec:
    ingress:
    - from:
      - namespaceSelector:
          matchLabels:
            kubernetes.io/metadata.name: openshift-virtualization-os-images
      - namespaceSelector:
          matchLabels:
            kubernetes.io/metadata.name: openshift-cnv
```
TBH, not sure what is the difference between these two formats functional-wise

Reply:
ah, good catch!

Reply:
Looking at https://kubernetes.io/docs/concepts/services-networking/network-policies/#behavior-of-to-and-from-selectors I think the two Ingress rules is the correct way for our use case since we want ingress traffic from both namespaces.

Reply:
Should be fixed with 067813d170f1b18bd48f71c82a9c420ce5d69b06